subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskCacheabilityResolver.java

```diff
@@ -49,6 +51,16 @@ public DefaultTaskCacheabilityResolver(RelativeFilePathResolver relativeFilePath
         @Nullable OverlappingOutputs overlappingOutputs
     ) {
         if (cacheIfSpecs.isEmpty()) {
+            if (task.getClass().isAnnotationPresent(DoNotCacheByDefault.class)) {
```

Comment:
We need some tests that `@DoNotCacheByDefault` can't be used with `CacheableTask` or `CacheableTransform`.

Reply:
FTR, we agreed that it's both okay to use them together, and to use `@DisableCachingByDefault` on its own.